porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -51,6 +51,9 @@ type PackageVariantReconciler struct {
 
 const (
 	workspaceNamePrefix = "packagevariant-"
+
+	ConditionTypeValid             = "Valid"
+	ConditionTypeDownstreamEnsured = "DownstreamEnsured"
```

Comment:
It seems like this might actually be describing whether the PackageVariant resource could be reconciled. Could we just use the condition types from kstatus: https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus#conditions ?

Reply:
I will use `Ready`, I think that will be better (esp since @johnbelamaric made the same comment above)

Reply:
Not sure. The kstatus and "traditional" Ready status conventions seem to be in direct opposition to one another. :(

Reply:
The `Ready` condition and `Reconciling` is essentially the same thing. The kstatus guidance was written back when the Kubernetes API conventions should adhere to abnormal-true polarity, i.e. conditions should have the value `True` for abnormal situations. For reconciling, we decided that a resource being fully reconciled is the "happy" scenario, so the condition should be `False` in this situation. This is why the kstatus guidance use `Reconciling` (rather than `Reconciled`) as the name. `Ready` is widely used and the API conventions were changed a couple of years ago, so I don't see any issue with using `Ready`.

The `Stalled` condition is there to distinguish between "not reconciled, but making progress" and "not reconciled and it is not making progress" (i.e. doomed). The idea is that `Stalled` being true means some kind of external action is probably required in order to make the resource reconcile. It seems like it might be similar to the use of the `Valid` condition here. Typically any resources that fail validation would need external action in order to make progress.

Reply:
Ok, so, it sounds like the summary is:
* Use `Ready` instead of `Reconciling`
* Use `Stalled` instead of `Valid`

Do we want to set `Reconciling` == !`Ready`, just in case there is tooling that cares? Or `Ready` is good enough?

In the #3910 changes, I want to document what we're actually going to do. There are new cases where `Stalled` is appropriate.

Note that @SimonTheLeg from Kubermatic and the Nephio community is going to pick up implementing the Kptfile pipeline editing feature described in #3910 so I want to get that buttoned up before I am off next week.

Reply:
> Ok, so, it sounds like the summary is:
> 
> Use Ready instead of Reconciling
> Use Stalled instead of Valid

I'm using ready atm so I will update `Valid` to `Stalled`. Seems to me like we don't need `Reconciling` in addition to `Ready` (and we could always add it later), but I'll defer to @mortent on this.

Reply:
Yeah, I think using `Stalled` instead of `Valid` seems correct. We can use the `reason` and `message` fields to communicate that the configuration is not valid.